### ISSUE TITLE
Consider changing default number of cores used 

### ISSUE THREAD

**Issue by maxim-h** [r-lib/pak#675]
Dear `pak` developers. Thank you for this cool project!

I just want to bring up a little pet peeve I often find with how software handles multi-core utilization. 

Spending a lot of time in shared environments makes me a bit wary of tools where utilizing all cores on the machine is the opt-out behavior. I can't even count the number of times that caused issues for either me or my colleagues.

One might argue that defining some global `Ncpus` value is a good practice in such environment in the first place. But in R I almost never encountered a need to use it (most packages treat parallelization as opt-in) and it's still a rake that many people might step on, especially since this behavior is explained as far as I can tell only on the website and [just a bit deeper](https://pak.r-lib.org/reference/pak-config.html) than the first time user would want to dig.

Would you consider changing the default behavior to NOT run in parallel unless `Ncpus` is explicitly set?

**Comment by gaborcsardi** [comment]
Considering that after many years, this is the first request for this, I think leaving it as opt out is the right decision. 

All machines are multicore nowadays, and 99.99% of the people are not using supercomputers.

**Comment by maxim-h** [comment]
I see your point. Just thought I'd bring it up. Feel free to close it if it's not planned.

Considering the lack of requests for many years, I don't know if most people in the ecosystem are quite aware of `pak` yet. But it feels like something that can be much more ubiquitous in the future, so the "demographic" could shift a bit as well.

P.S.: Love the `renv` compatibility. It's lack was stopping me from trying `pak` before. Hope the smaller issues can get ironed out.

**Comment by timothee-bacri** [comment]
@gaborcsardi Note that using all cores is bad practice, as it can freeze the entire computer and prevent moving the mouse, responding to keyboard presses, updating the desktop clock. I have run into these limitations.

The better way is to leave at least 1 or 2 cores available for system tasks.

Maybe consider giving this a low priority if you want, but closing it is not ideal.
